internal/store/device.go

```diff
@@ -190,6 +240,18 @@ func (s *DeviceStore) Delete(ctx context.Context, orgId uuid.UUID, name string)
 	return result.Error
 }
 
+func (s *DeviceStore) UpdateDeviceOwner(ctx context.Context, orgId uuid.UUID, name string, owner string, callback DeviceStoreCallback) error {
+	device := model.Device{Resource: model.Resource{OrgID: orgId, Name: name}}
+	err := s.db.Model(device).Updates(map[string]interface{}{
+		"generation": 0,
+		"owner":      owner,
+	}).Error
```

Comment:
I would also insert a status condition update about the owner change for some degree of traceability on the device.

Reply:
What kind of condition?  When would it be set/unset?
Also, the agent owns the device status.
I was thinking that we would implement an event mechanism for traceability in general.